### PR TITLE
Support color correction for img2img and inpainting

### LINKED ISSUE #552: Remove magenta hue from img2img outputs
Using img2img and looping the image back through it causes an increase in magenta hues and saturation of the output image. I think they used this fix https://github.com/Pyr-000/YetAnotherStableDiffusion/commit/b8751862de557edd0cad2d9f7e4b4e46c3f3042c in the Automatic1111 fork to fix this and it works pretty good so far from my limited testing.  Would be a nice to add this or something similar.

### PR DESCRIPTION
**Pull request by db3000**
Fixes #552

Perform image color correction, borrowed from https://github.com/Pyr-000/YetAnotherStableDiffusion/commit/c7bf958c2dfa88fc027690983fee1f877e8827e9

Note that the correction needs to be done against a stable initial image when running img2img or inpainting multiple times. If the correction is done only against the previous image, you end up with everything turning cyan, as in https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/481

Using tests from that ticket, before correction:
out2

After correction:
out

### CODE REVIEW

**Review by lstein — CHANGES_REQUESTED** [review]
This looks fine, but recent changes to how the command-line arguments are stored and parsed prevent it from being merged automatically. The argument parsing now lives in ldm/dream/args.py If you can, could you rebase against current development and resolve the conflicts?

Thanks!

**Comment by db3000** [comment]
Thanks, rebased against development. Seems my test script needed to be changed as it was hitting a new issue: https://github.com/lstein/stable-diffusion/issues/628

**Comment by lstein** [comment]
#628 is fixed by PR #654, which should go in very soon. I'll run some testing on your PR tomorrow (Sunday) and if all goes well it'll go into dev. Thanks so much for the rebasing!

File: backend/modules/parameters.py [hunk 1]
```diff
@@ -40,6 +40,8 @@ def parameters_to_command(params):
         switches.append(f'-I {params["init_img"]}')
     if 'init_mask' in params and len(params['init_mask']) > 0:
         switches.append(f'-M {params["init_mask"]}')
+    if 'init_color' in params and len(params['init_color']) > 0:
```
Comment (lstein) [inline backend/modules/parameters.py:43]: Aside for @Kyle0654: Here's an example of a PR that has had to make parallel changes among the web and CLI interfaces in order to introduce a new switch. I'm looking forward to you unifying the command-line argument processing in the new architecture.

**Review by lstein — APPROVED** [review]
It seems to be working as advertised. 

In general, the quality of the images is getting degraded with multiple passes through img2img, and I wonder if this reflects over-aggressive compression when writing out the PNGs. Better still would be for a way to keep the uncompressed image in memory while operating on it, rather than writing it out to disk each time.

**Comment by db3000** [comment]
Thanks! Yeah I saw that as well, I thought that was further artifacts of the diffusion process. Could validate that by skipping the actual processing and seeing if the artifacts appear still?

**Pull request merged**